modules/nextflow/src/main/groovy/nextflow/container/DockerBuilder.groovy

```diff
@@ -117,7 +131,10 @@ class DockerBuilder extends ContainerBuilder<DockerBuilder> {
         result << 'run -i '
 
         if( cpus ) {
-            if( legacy )
+
+            result << "--cpus ${countCpus(cpus)} "
+
+            if (legacy)
```

Comment:
@pditommaso , I've tried to accomodate the newer `--cpus` option with the `cpuset`. The `countCpus` private method (defined above) computes it from the `cpus`.

Reply:
Hold-on. These are two different things. Much easier to rename the old `cpus` to `cpuset`, and add a new attribute `Float cpus` . The option `--cpus <cpus>` should be formatted to one decimal precision digit.